src/main/java/fr/quatrevieux/araknemu/game/fight/turn/action/move/MoveStatus.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * This file is part of Araknemu.
+ *
+ * Araknemu is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Araknemu is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Araknemu.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Copyright (c) 2017-2020 Vincent Quatrevieux
+ */
+
+package fr.quatrevieux.araknemu.game.fight.turn.action.move;
+
+import fr.arakne.utils.maps.constant.Direction;
+import fr.arakne.utils.maps.path.Path;
+import fr.quatrevieux.araknemu.game.fight.fighter.Fighter;
+import fr.quatrevieux.araknemu.game.fight.map.FightCell;
+import fr.quatrevieux.araknemu.game.fight.turn.action.ActionResult;
+
+/**
+ * Successful move result
+ */
+abstract public class MoveStatus implements ActionResult {
```

Comment:
Abstract class is not relevant : use an interface instead, no code are shared between MoveSuccess and MoveFailed